core/src/main/scala/com/eharmony/spotz/optimizer/StopStrategy.scala

```diff
@@ -0,0 +1,85 @@
+package com.eharmony.spotz.optimizer
+
+import org.joda.time.Duration
+
+/**
+  * @author vsuthichai
+  */
+sealed trait StopStrategy extends Serializable {
+  final val UNLIMITED = Long.MaxValue
+  final val FOREVER = new Duration(Long.MaxValue)
+
+  def getMaxTrials: Long = UNLIMITED
```

Comment:
I've left the getMaxTrials accessor in there, but deleted the other one.  It does play a part in determining batch size.